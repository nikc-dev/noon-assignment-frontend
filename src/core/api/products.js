import axios from 'axios'
const endPoint = 'http://127.0.0.1:8000/'


export const getAllProducts = () => {
    return axios.get(endPoint)
}

export const updateLikedProducts = (payload) => {
    return axios.post(endPoint, payload)
}

export const getLikedProducts = () => {
    return axios.get(`${endPoint}favs`)
}

export const removeLikedProducts = (payload) => {
    return axios.delete(`${endPoint}favs`, {headers: {}, data : payload })
}