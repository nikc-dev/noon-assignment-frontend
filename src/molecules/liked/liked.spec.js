import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from 'enzyme-to-json';
import Liked from './index';
import * as productApi from '../../core/api/products'

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

beforeEach(() => {
  productApi.getLikedProducts = jest.fn(() => Promise.resolve({data:[]}));
});

it('renders correctly', () => {
  const component = renderer.create(
    <Liked />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render liked products with api response', () => {
  productApi.getLikedProducts = jest.fn(() => Promise.resolve({data:[{ 'id': '001', 'productName': 'test', 'productDescription': 'test', 'liked': true }]}));
  const component = shallow(
    <Liked />,
  );
  expect(component.state('products')).toBeUndefined();
  expect(toJson(component)).toMatchSnapshot();
});

it('should render error with api empty response', () => {
  productApi.getLikedProducts = jest.fn(() => Promise.resolve({data:[]}));
  const component = shallow(
    <Liked />,
  );
  expect(component.state('isError')).toBeFalsy();
  expect(toJson(component)).toMatchSnapshot();
});
