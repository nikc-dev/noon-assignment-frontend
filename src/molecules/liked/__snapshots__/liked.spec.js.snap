// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="liked"
>
  <div
    className="loader"
  />
</div>
`;

exports[`should render error with api empty response 1`] = `
<div
  className="liked"
>
  <Loader />
</div>
`;

exports[`should render liked products with api response 1`] = `
<div
  className="liked"
>
  <Loader />
</div>
`;
